src/services/lnd/index.ts

```diff
@@ -256,6 +254,81 @@ export const LndService = (): ILightningService | LightningServiceError => {
     return new PaymentNotFoundError("Payment hash not found")
   }
 
+  const listFailedPayments = async ({
+    after,
+    pubkey,
+  }: {
+    after: PagingStartToken | PagingContinueToken
+    pubkey: Pubkey
+  }): Promise<ListLnPaymentsResult | LightningServiceError> => {
+    const { lnd } = getLndFromPubkey({ pubkey })
+    const pagingArgs = after ? { token: after } : {}
+
+    try {
+      const { payments, next } = await getFailedPayments({ lnd, ...pagingArgs })
+      return {
+        lnPayments: payments
+          .map(translateLnPaymentLookup)
+          .map((p) => ({ ...p, status: PaymentStatus.Failed })),
+        endCursor: (next as PagingContinueToken) || false,
+      }
+    } catch (err) {
+      return new UnknownLightningServiceError(err)
+    }
+  }
+
+  const listSettledPayments = async ({
+    after,
+    pubkey,
+  }: {
+    after: PagingStartToken | PagingContinueToken
+    pubkey: Pubkey
+  }): Promise<ListLnPaymentsResult | LightningServiceError> => {
+    try {
+      const { lnd } = getLndFromPubkey({ pubkey })
+      const pagingArgs = after ? { token: after } : {}
+      const { payments, next } = await getPayments({ lnd, ...pagingArgs })
+
+      return {
+        lnPayments: payments
+          .map(translateLnPaymentLookup)
+          .filter((p) => p.status === PaymentStatus.Settled),
+        endCursor: (next as PagingStartToken) || false,
+      }
+    } catch (err) {
+      return new UnknownLightningServiceError(err)
+    }
+  }
+
+  const listSettledAndFailedPaymentsByPubkey = async (
```

Comment:
the other method also take a pubkey not sure we need to add this in the name of the function